### PR TITLE
Malf AI Changes

### PR DESCRIPTION
**Pull request by SarmentiCampbell**
## About The Pull Request
Basically, Transformers, The autoborger perma kill peoples, even when you take out the brain and clone it, It would turn into a generic body, I haven't found a way to fix it even using mmi.transfer.identity. I tried to find a way for the A.I to be able to receive his malf objectives instead of always getting the 'teach someone a lesson' objective even if it is shitty. in 6 AM, sob

## Why It's Good For The Game
Nerfed malf ai modules, removed transformers and modified a few objectives

## Changelog
:cl:
add: Added a shitty way for ai to get his objectives
del: Removed transformer, Malf AI Module
tweak: tweaked machine overload's explosion and machine's overdrive's uses
del: Decided to remove my brain for this. It is way too shitty.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -343,7 +343,7 @@ GLOBAL_LIST_EMPTY(objectives)
 
 /datum/objective/block
 	name = "no organics on shuttle"
-	explanation_text = "Do not allow any organic lifeforms to escape on the shuttle alive."
+	explanation_text = "Do not allow any organic life forms aboard the shuttle." //Skyrat Change
```
Comment (Akrilla) [inline code/game/gamemodes/objective.dm:346]:
```suggestion
	explanation_text = "Do not allow any organic lifeforms aboard the shuttle." //Skyrat Change
```

File: code/modules/antagonists/traitor/classes/ai.dm [hunk 2]
```diff
@@ -44,7 +31,30 @@
 			yandere_two.target = yandere_one.target
 			yandere_two.update_explanation_text() // normally called in find_target()
 			T.add_objective(yandere_two)
-			.=2
+		if(5) // Skyrat Change - I'm just gonna add in another yandere objective here and increase the chance to roll it because yandere ai is  y e s.
+			var/datum/objective/protect/yandere_one = new
```
Comment (Akrilla) [inline code/modules/antagonists/traitor/classes/ai.dm:35]: I think I want this as its own objective e.g. datum/objective/yandere

File: code/modules/antagonists/traitor/equipment/Malf_Modules.dm [hunk 3]
```diff
@@ -656,80 +656,7 @@ GLOBAL_LIST_INIT(blacklisted_malf_machines, typecacheof(list(
...
-				success = FALSE
-		var/image/I = action.turfOverlays[n]
-		I.loc = T
-		client.images += I
-		I.icon_state = "[success ? "green" : "red"]Overlay" //greenOverlay and redOverlay for success and failure respectively
-		addtimer(CALLBACK(src, .proc/remove_transformer_image, client, I, T), 30)
-	if(!success)
-		to_chat(src, "<span class='warning'>[alert_msg]</span>")
-	return success
-
```
Comment (Afya212) [inline code/modules/antagonists/traitor/equipment/Malf_Modules.dm:732]: This all should be commented out, not straight up deleted, especially considering we're a downstream.

File: code/modules/antagonists/traitor/classes/ai.dm [hunk 4]
```diff
@@ -1,40 +1,27 @@
 /datum/traitor_class/ai // this one is special, so has no weight
 	name = "Malfunctioning AI"
+	employer = "ERROR ER0RR $R0RRO$"
```
Comment (Afya212) [inline code/modules/antagonists/traitor/classes/ai.dm:3]:
```suggestion
	employer = "ERROR ER0RR $R0RRO$" //Skyrat change
```

File: code/modules/antagonists/traitor/classes/ai.dm [hunk 5]
```diff
@@ -1,40 +1,27 @@
...
-
-	for(var/i = objective_count, i < CONFIG_GET(number/traitor_objectives_amount), i++)
-		var/datum/objective/assassinate/kill_objective = new
-		kill_objective.owner = T.owner
-		kill_objective.find_target()
-		T.add_objective(kill_objective)
-
-	var/datum/objective/survive/exist/exist_objective = new
-	exist_objective.owner = T.owner
-	T.add_objective(exist_objective)
```
Comment (Afya212) [inline code/modules/antagonists/traitor/classes/ai.dm:19]: Big deletion should just be commented out since we're a downstream.

**Comment by YakumoChen** [comment]
:-1: on the overload and override changes. I remember old overload. Major 1 means the target has to be literally standing on the tile, ignore the
### **LOUD RED MESSAGE**
for three full seconds, and then it maybe crits them. 2 range means it hits in a 3x3 which is Actually Damaging and a decent threat. 

Override is just a pitiful power that is only good for getting rid of important machines, it turns them into slow, mob AI monsters that aren't really threatening, reducing uses to 2 neuters it badly.

**Comment by ForrestWick** [comment]
Malfs are already weak enough as is, they don't need any further nerfs. We have respawn enabled, there are mutation toxins, pod cloning exists, admin intervention exists, there is no such thing as perma killing. This kind of hugboxing is needless and kills any hope for an antagonist with crapshoot odds and a requirement to be killed unless admins step in if you want to prevent the threat entirely.

If you want to change Malfs, then make a rework, not needless nerfs and removals.

**Comment by ShadeAware** [comment]
Breaching entire rooms with a power you can spam across the entire station, a **dirt cheap** power at that, isn't exactly what I would call good gameplay for anyone involved @YakumoChen. You shouldn't be able to vent an entire room just by going "haha machine button go bbrrrrrr", and then vent the entire station because almost ANYTHING can be overridden. I've seen pipes, scrubbers, fucking _lights_ made to explode with the power of a breaching charge.

**Comment by SarmentiCampbell** [comment]
Okay, Come again and tell me how a skill that cost only 20 processing time which can cause breach and be brought multiple time was good again?
I could understand that it was a tiny bit excessive for overdrive but overload is way too cheap and deserve to be nerfed.

Also, Transformer is broken due to the fact that you are unable to get your actual body cloned back.
As in, Taking the MMI out of the cyborg and putting the brain into a clone, which clone a generic human, not your character.

**Comment by YakumoChen** [comment]
> Breaching entire rooms with a power you can spam across the entire station, a **dirt cheap** power at that, isn't exactly what I would call good gameplay for anyone involved @YakumoChen. You shouldn't be able to vent an entire room just by going "haha machine button go bbrrrrrr", and then vent the entire station because almost ANYTHING can be overridden. I've seen pipes, scrubbers, fucking _lights_ made to explode with the power of a breaching charge.

Breaches are a lot less deadly than you make them to be. They are fairly easy to run away from and easy to patch if you work fast. Most of the damage of overload is the initial explosion, and a one tile breach is almost a nil threat even if you avoid the damage portion of the explosion. 
AI has very few means itself of directly interacting with the crew, especially as traitor, and Overload is perhaps the only actually viable means of doing so that isn't a shocked airlock (which only affects 2/3rds of the crew at all or widespread damage ie a plasma flood. You could bolt them into a siphoning room, but this is pretty inefficient, time consuming, and very easily undone. and for a malf AI, time is _EVERYTHING_.

> Okay, Come again and tell me how a skill that cost only 20 processing time which can cause breach and be brought multiple time was good again?
I could understand that it was a tiny bit excessive for overdrive but overload is way too cheap and deserve to be nerfed.

20 is... a surprising amount of cycles. There's not a lot of APCs you can safely hack without getting caught.

Remember that a traitor AI, despite being on an armed satellite, is actually fairly easy to kill if you know what you're doing. A single large EMP in space just outside the core is usually enough to knock out the AI completely, or gives you a free pass to breach the chamber and finish it off.

**Comment by SarmentiCampbell** [comment]
Honestly, We all know that Malf AI need a rework, But I ain't the guy that can do it, But for now, I am just fixing the objectives and nerfing his kit a tiny bit for the sole reason that peoples would continue complaining and would very fucking soon make them remove malf ai for 'lrp'

**Comment by YakumoChen** [comment]
Nerfing to unviability is no better than full removal. The argument goes both ways. While the AI is a very powerful traitor, it has similarly extreme Achilles Heels.

**Comment by Azarak** [comment]
Malf AI would require a much thorough look and adjustments, both in objectives and tools and metaknowledge. Shifts with Malf AI tend to be very polarized due to the underlying designs and conflicts with our rules. I do feel like the other malf objectives are not gonna make it any better. Yendere, Blocker and Purge objectives are much like the traitor's Maroon objective, basically makes the AI act radically just at the shift end while not doing anything interesting throught the rest of it.

**Comment by jjpark-kb** [comment]
Turning stale. A larger rework of AI may be necessary.

**Pull request closed without merging**